File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/pool/basicfunctionality/item/PoolItemSettings.java
package ru.urfu.javapools.poolslibrary.pool.basicfunctionality.item;

public class PoolItemSettings<TK> {
	
	private TK _key;
	private int _maxObjectsCount;
	private boolean _markObtainedObjectAsNotAvailable;
	
	public TK getKey() {
		return _key;
	}

	public boolean getMarkObtainedObjectAsNotAvailable() {
		return _markObtainedObjectAsNotAvailable;
	}
	
	public int getMaxObjectsCount() {
		return _maxObjectsCount;
	}

	public PoolItemSettings<TK> setKey(TK key) {
		_key = key;
		return this;
	}

	public PoolItemSettings<TK> setMaxObjectsCount(int maxObjectsCount) {
		_maxObjectsCount = maxObjectsCount;
		return this;
	}

	public PoolItemSettings<TK> setMarkObtainedObjectAsNotAvailable(boolean markObtainedObjectAsNotAvailable) {
		_markObtainedObjectAsNotAvailable = markObtainedObjectAsNotAvailable;
		return this;
	}	
}

File: CSharpPools/PoolsLibrary/Pool/BasicFunctionality/Item/PoolItemSettings.cs
namespace PoolsLibrary.Pool.BasicFunctionality.Item
{
    class PoolItemSettings<TK>
    {
        public TK Key { get; set; }
        public bool MarkObtainedObjectAsNotAvailable { get; set; }

        public int MaxObjectsCount { get; set; }
        public bool ThrowIfCantCreateNewBecauseOfReachedLimit { get; set; }
    }
}
